types/babel__core/index.d.ts

```diff
@@ -528,4 +554,129 @@ export interface CreateConfigItemOptions {
  */
 export function createConfigItem(value: PluginTarget | [PluginTarget, PluginOptions] | [PluginTarget, PluginOptions, string | undefined], options?: CreateConfigItemOptions): ConfigItem;
 
+// NOTE: the documentation says the ConfigAPI also exposes @babel/core's exports, but it actually doesn't
+/**
+ * @see https://babeljs.io/docs/en/next/config-files#config-function-api
+ */
+export interface ConfigAPI {
+    /**
+     * The version string for the Babel version that is loading the config file.
+     *
+     * @see https://babeljs.io/docs/en/next/config-files#apiversion
+     */
+    version: string;
+    /**
+     * @see https://babeljs.io/docs/en/next/config-files#apicache
+     */
+    cache: SimpleCacheConfigurator;
+    /**
+     * @see https://babeljs.io/docs/en/next/config-files#apienv
+     */
+    env: EnvFunction;
+    // undocumented; currently hardcoded to return 'false'
+    // async(): boolean
+    /**
+     * This API is used as a way to access the `caller` data that has been passed to Babel.
+     * Since many instances of Babel may be running in the same process with different `caller` values,
+     * this API is designed to automatically configure `api.cache`, the same way `api.env()` does.
+     *
+     * The `caller` value is available as the first parameter of the callback function.
+     * It is best used with something like this to toggle configuration behavior
+     * based on a specific environment:
+     *
+     * @example
+     * function isBabelRegister(caller?: { name: string }) {
+     *   return !!(caller && caller.name === "@babel/register")
```

Comment:
Good: `@example` actually highlights whatever comes after it as typescript code, but without the weird bug it has with markdown code fences

Bad: the `@` in `@babel/` breaks it in tooltips.

Code | Tooltip
--- | ---
![image](https://user-images.githubusercontent.com/73085/52902628-deea8b00-3256-11e9-922e-e74871de3ff4.png) | ![image](https://user-images.githubusercontent.com/73085/52902638-fe81b380-3256-11e9-887e-c1bff66104b7.png)

Alternative with code fences:

Code | Tooltip
--- | ---
![image](https://user-images.githubusercontent.com/73085/52902644-11948380-3257-11e9-954a-b46339dfebbb.png) | ![image](https://user-images.githubusercontent.com/73085/52902649-1bb68200-3257-11e9-920f-72634d356544.png)